tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1251,5 +1251,31 @@ def test_assertion_is_failure(dut):
     assert False
 
 
+@cocotb.test(expect_error=True)
+def test_timeout_func_fail(dut):
+    @cocotb.coroutine
+    def fail():
+        yield Timer(10, 'ns')
+    yield cocotb.triggers.with_timeout(fail(), timeout_time=1, timeout_unit='ns')
+
+
+@cocotb.test(expect_error=False)
+def test_timeout_func_pass(dut):
+    @cocotb.coroutine
+    def success():
+        yield Timer(10, 'ns')
+    yield cocotb.triggers.with_timeout(success(), timeout_time=100, timeout_unit='ns')
```

Comment:
Please add a return value to `success() here, to test the bug I identified in https://github.com/cocotb/cocotb/pull/1119/files#r346068338